src/main/java/com/backend/entities/TaskCompletionRecord.java

```diff
@@ -1,27 +1,93 @@
 package com.backend.entities;
 
-import com.backend.entities.IDs.AccountID;
 
-import java.sql.Timestamp;
+import com.backend.entities.IDs.AccountID;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.PersistenceCreator;
+import org.springframework.data.annotation.Transient;
+import org.springframework.data.mongodb.core.mapping.Document;
 
+/**
+ * completion record for tasks
+ */
+@Document(collection = "TaskCompletedCollection")
 public class TaskCompletionRecord {
-    private final AccountID accountID;
-    private final Timestamp timestamp;
-    private final String taskName;
+    //instance variables
+    @SuppressWarnings("unused")
+    @Id
+    private String ID;
+    @Transient
+    private final AccountID accountIDObject;
+    private final String accountID;
+    private final String timestamp;
```

Comment:
change this from java.sql.timestamp to java.util.Date

Reply:
It's a String